utils/proxy/proxyconfig.go

```diff
@@ -115,6 +114,9 @@ func (pc *ProxyConfig) useProxy(addr string) bool {
 			// no_proxy "foo.com" matches "bar.foo.com"
 			return false
 		}
+		if _, net, err := net.ParseCIDR(p); ip != nil && err == nil && net.Contains(ip) {
```

Comment:
shouldn't you check ip != nil before you bother to ParseCIDR ? Given that if the original ip is nil, then the net.Contains doesn't make any sense. Maybe something like:
if ip != nil {
  // this is an IP, check to see if we were given a CIDR instead of just an exact IP match
}